domain/blockdag/validate.go

```diff
@@ -963,8 +963,8 @@ func checkEntryAmounts(entry *UTXOEntry, totalSompiInBefore uint64) (totalSompiI
 	lastSompiIn := totalSompiInBefore
 	originTxSompi := entry.Amount()
 	totalSompiInAfter = totalSompiInBefore + originTxSompi
-	if totalSompiInBefore < lastSompiIn ||
-		totalSompiInBefore > util.MaxSompi {
+	if totalSompiInAfter < lastSompiIn ||
+		totalSompiInAfter > util.MaxSompi {
```

Comment:
`lastSompiIn` can be removed and we can check against `totalSompiInBefore` directly